tools/make_patch.c

```diff
@@ -0,0 +1,597 @@
+			strcpy(searchend, current_hook->name);
+			strcat(searchend, "_End");
+			current_hook = find_symbol(symbols, searchend);
+			if (!current_hook) {
+				fprintf(stderr, "Error: Could not find symbol: %s", searchend);
+			} else {
+				// Figure out the length of the patch
+				int length = current_hook->offset - offset;
+				memset(searchend, 0, (strlen(current_hook->name)));
+				// We've got the length, now go back
+				fseek(new_rom, offset, SEEK_SET);
+				// Print out the patch
+				append_patch(patches, offset, length);
+				fprintf(output, "a%d:", length);
+				for (int i = 0; i < length; i++) {
+					if (i) {
+						putc(' ', output);
+					}
+					fprintf(output, isupper((unsigned char)command[0]) ? "%02X" : "%02x", getc(new_rom));
+				}
+			}
+		}
+
+	} else if (!strcmp(command, "Constant") || !strcmp(command, "constant")) {
+		if (argc != 2) {
+			fprintf(stderr, "Error: Missing argument for %s", command);
+		}
+		const struct Symbol *getsymbol = find_symbol(symbols, argv[1]);
+		if (!getsymbol) {
+			return;
+		}
+		int parsed_offset = getsymbol->address;
+		if (!strcmp(argv[0], "db")) {
+			fprintf(output, isupper((unsigned char)command[0]) ? "%02X": "%02x",
+				parsed_offset);
+		} else {
+			fprintf(output, isupper((unsigned char)command[0]) ? "%02X %02X": "%02x %02x",
+				parsed_offset, HIGH(parsed_offset));
+		}
+
+	} else if (!strcmp(command, "findaddress")) {
+		if (argc != 1) {
+			fprintf(stderr, "Error: Missing argument for %s", command);
+		}
+		const struct Symbol *getsymbol = find_symbol(symbols, argv[0]);
+		if (!getsymbol) {
+			return;
+		}
+		fprintf(output, "0x%x", getsymbol->offset);
+
+	} else if (!strcmp(command, "conaddress")) {
+		if (argc != 2) {
+			fprintf(stderr, "Error: Missing argument for %s", command);
+		}
+		const struct Symbol *getsymbol = find_symbol(symbols, argv[1]);
+		if (!getsymbol) {
+			return;
+		}
+		int parsed_offset = getsymbol->address;
+		if (!strcmp(argv[0], "dw")) {
+			fprintf(output, "%02x ", LOW(parsed_offset));
+			fprintf(output, "%02x ", HIGH(parsed_offset));
+			fprintf(output, "%02x ", LOW(parsed_offset + 1));
+			fprintf(output, "%02x",  HIGH(parsed_offset + 1));
+		} else {
+			fprintf(output, "%02x ", LOW(parsed_offset));
+			fprintf(output, "%02x",  HIGH(parsed_offset));
+		}
+
+	} else if (!strcmp(command, "==")) {
+		fprintf(output, "00 00");
+	} else if (!strcmp(command, ">")) {
+		fprintf(output, "01 00");
+	} else if (!strcmp(command, "<")) {
+		fprintf(output, "02 00");
+	} else if (!strcmp(command, ">=")) {
+		fprintf(output, "03 00");
+	} else if (!strcmp(command, "<=")) {
+		fprintf(output, "04 00");
+	} else if (!strcmp(command, "!=")) {
+		fprintf(output, "05 00");
+
+	} else {
+		fprintf(stderr, "Error: Unknown command: %s\n", command);
+	}
+}
+
+struct Patches *process_template(
+	const char *template_filename, FILE *new_rom, FILE *orig_rom,
+	const char *output_filename, const struct Symbol *symbols
+) {
+	FILE *file = xfopen(template_filename, 'r');
```

Comment:
```suggestion
	FILE *template = xfopen(template_filename, 'r');
```

Reply:
Resolved in https://github.com/pret/pokecrystal/pull/882/commits/e58df7e55f4239e532524664a8cf9ab00268f876